catalogue/webapp/utils/dates.ts

```diff
@@ -52,17 +52,29 @@ export function convertDayNumberToDay(dayNumber: DayNumber): Day {
   }
 }
 
-export function findNextRegularOpenDay(
+export function findNextPickUpDay(
   date: Moment,
   regularClosedDays: DayNumber[]
 ): Moment | undefined {
   if (regularClosedDays.length === 7) {
     // All days are closed, so we'll never be able to find a non closed day.
     return undefined;
   }
-  const isClosed = regularClosedDays.includes(date.day() as DayNumber);
-  if (isClosed) {
-    return findNextRegularOpenDay(date.add(1, 'day'), regularClosedDays);
+
+  // If the library is closed on this day, we want to set the pick-up day to be
+  // the next open day plus one, so that e.g. Monday morning isn't a scramble
+  // for library staff handling the weekend's requests. Since this function
+  // calls itself recursively, we add one day if we're closed today and
+  // tomorrow, but add two days if we're closed today and open tomorrow.
```

Comment:
nit:

```suggestion
  // tomorrow, but add two days if we're closed on this day and open the next.
```

because the today/tomorrow naming was confusing me yesterday